### ISSUE TITLE
Full Accessibility & 508 Office Audit [Profile 2.0]

### ISSUE THREAD

**Issue by mshea0606** [department-of-veterans-affairs/va.gov-team#15247]
## Steps to complete the Full Accessibility & 508 Audit: 
- [x] Product Manager: create issue and fill in feature name in the title and other bolded information appropriately
- [ ] Practice area reviewers create sub-issues attached to this one with feedback for your team on *feature name*
- [ ] **Requesting team** completes attached feedback tickets based on definition of done included in each practice area feedback ticket (may differ for each practice area, please refer to attached tickets for further instruction)
- [ ] All requesting team attendees complete brief [VSP Collaboration Cycle Feedback](https://adhoc.optimalworkshop.com/questions/20260uu8-0-0/questions/before) survey

## Artifacts - _please provide the following artifacts when you request a full review_
- Access information for the tool or feature in staging : [Staging Credentials](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/products/identity-personalization/Staging%20Credentials.md)
  - Complete/check-off the required items here:
    - [x] [Axe scans in daily development](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#axe-scans-in-daily-development)
    - [x] [Axe scans in end-to-end tests](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#axe-scans-in-end-to-end-tests)
    - [x] [Color tests](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#color-tests)  
    - [x] [Content resize check](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#content-resize-check)
    - [x] [Keyboard navigation check](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#keyboard-navigation-check)
    - [x] [Screen reader tests](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/staging-review-processes.md#screen-reader-tests)

## Additional Notes
As part of this review, VSP will send your product to VA's 508 office for approval (as applicable)

**Comment by jenstrickland** [comment]
Thanks @mshea0606! We have a backlog of full accessibility audits right now, but I will get to this ASAP.

**Comment by mshea0606** [comment]
thanks @jenstrickland!

**Comment by 1Copenut** [comment]
@mshea0606 I'm not seeing any issues attached to this audit. Has it been completed, or should I reopen it and work with the VSA specialists to get it started?

**Comment by 1Copenut** [comment]
Reopening issue after discussing with a11y specialists and VSA design lead.

**Comment by joshkimux** [comment]
@ajakabcin Angela will be taking up this review!

**Comment by Samara-Strauss** [comment]
Thanks, @joshkimux! Do we have an estimate on when we should get results back, or is that kind of hard to say right now? I'm mostly curious, so no worries on whatever the answer/timeline is.

**Comment by AngelaFowler82** [comment]
My goal is to have it done by the end of tomorrow.

**Comment by Samara-Strauss** [comment]
Thanks for letting us know!

**Comment by joshkimux** [comment]
@Samara-Strauss @AngelaFowler82 
Given that we're still catching up on other full a11y audits, there is not hard deadline for us to get this back. Please feel free to work on this based on your availability!

**Comment by joshkimux** [comment]
@Samara-Strauss @AngelaFowler82 
I just reviewed the test case document and there appears to be 29 different logins that we may need to review. I'm guessing that this will go by quickly since there will be a good amount of overlap, but I think it's fair to estimate out this review for at least 1-2 weeks to complete.

**Comment by Samara-Strauss** [comment]
Sounds good! Thank you!

**Comment by AngelaFowler82** [comment]
Hello,

Would someone be so kind as to send Josh and I the password for all the test accounts? We are unable to find them.

Angela

**Comment by Samara-Strauss** [comment]
Password info is in the private repo:

- [Master list of staging users](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/mvi-staging-users.csv)
- [Staging users folder](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/tree/master/Administrative/vagov-users)

**Comment by joshkimux** [comment]
@AngelaFowler82 here's the username and passwords of folks we'll be testing for

vets.gov.user+1@gmail.com
213SsNrLgPv5

vets.gov.user+2@gmail.com
194SsNrLgPv5

vets.gov.user+10@gmail.com
427SsNrLgPv5

vets.gov.user+27@gmail.com
469SsNrLgPv5

vets.gov.user+36@gmail.com
419SsNrLgPv5

vets.gov.user+202@gmail.com
220SsNrLgPv5

vets.gov.user+226@gmail.com
318SsNrLgPv5

@Samara-Strauss , I couldn't find the passwords for the following users. Any chance you could help us here? 🙏 
vets.gov.user+350@gmail.com
vets.gov.user.555@gmail.com

**Comment by Samara-Strauss** [comment]
@joshkimux I hae deleted your post for security reasons. We can not have password info in the public repo.

**Comment by AngelaFowler82** [comment]
Can we create a private document with the specific user accounts and passwords you would like tested? It would save us considerable time hunting for passwords and matching them with use cases. There are a lot of test accounts!

**Comment by joshkimux** [comment]
@Samara-Strauss @AngelaFowler82 I'm on it! I'll attach a link in a jiffy :)

**Comment by joshkimux** [comment]
@AngelaFowler82 Here's the [profile specific username and password list within the sensitive repo!](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/Administrative/vagov-users/staging-test-accounts-profile)

@Samara-Strauss Would love your help populating user 350 and 555 should you have the time! 🙇

**Comment by Samara-Strauss** [comment]
@joshkimux @AngelaFowler82 unfortunately, we need to have all of those passwords reset now because they've been compromised by being put in the public repo. I am working on that and will update documentation when I have new passwords.

**Comment by joshkimux** [comment]
Thank you!

**Comment by Samara-Strauss** [comment]
Documentation has been updated

**Comment by Samara-Strauss** [comment]
> @Samara-Strauss Would love your help populating user 350 and 555 should you have the time! 🙇

- User 350 used to be our only LOA1 staging user, but it looks like it got upgraded to LOA3. This means that for all LOA1 use cases, Angela will have to create an account on staging to view these use cases. **It's important to note that when this new user gets created, you should not verify it's identity or add 2FA in the sign up process.** This will allow you to create the necessary use cases.
- I couldn't find a password for user 555, but I removed this test case — which was to see what it looks like when someone has added 2FA — because an LOA1 and LOA3 user will see the same thing, and we already have a test case for this with an LOA3 user.

**Comment by joshkimux** [comment]
<img width="694" alt="Screen Shot 2021-04-14 at 5 04 26 PM" src="https://user-images.githubusercontent.com/14154792/114778941-7a96b980-9d43-11eb-920c-7a33a2a2d71f.png">

Are folks unable to specify their gender?

**Comment by Samara-Strauss** [comment]
@joshkimux yes, currently they are. This gets pulled in from MPI and is not editable from the interface at the moment.

**Comment by joshkimux** [comment]
Another design question--

<img width="976" alt="Screen Shot 2021-04-14 at 5 26 57 PM" src="https://user-images.githubusercontent.com/14154792/114781450-acf5e600-9d46-11eb-86cd-2695e6e96596.png">
<img width="821" alt="Screen Shot 2021-04-14 at 5 27 04 PM" src="https://user-images.githubusercontent.com/14154792/114781451-acf5e600-9d46-11eb-9c9e-3fb7c428d20d.png">

Is there a design reason for not making the first of these two modal? They share the same style, functions, and heading level, so I'm wondering if the former was meant to be modal?

Otherwise, since the former is non-modal and introduces a `h3` which isn't logically placed since "Home" is not a heading itself, we would need to change it to a `p` or convert it to a `h4` and bump "Home" up to a `h3`.

**Comment by AngelaFowler82** [comment]
A couple of rookie questions:

1.	What are the differences between loa1, loa2 and loa3 users? I know there’s information in the attached document, but its sorted by category rather than use case, and the same information seems to apply to multiple users making it difficult to interpret the information.
2.	When creating an account, do I use ID.Me, or something else?

Huge thanks.

**Comment by Samara-Strauss** [comment]
@joshkimux for design questions, please ask @tressaellen.

@AngelaFowler82 

> 1.	What are the differences between loa1, loa2 and loa3 users? I know there’s information in the attached document, but its sorted by category rather than use case, and the same information seems to apply to multiple users making it difficult to interpret the information.

Great question.

- LOA1 users have signed up for an account but have not verified their identity on VA.gov. This means we can't show them most of the information in the profile because without having verified their identity, the VA doesn't really know who someone is.
- LOA3 users have signed up for an account and have also verified their identity with the VA. This means we trust the person is who they say they are, and we can connect them to various backends that will show personalized information in the VA.gov interface.
- I am unfamiliar with LOA2; to the best of my knowledge, we only treat people as either LOA1 or LOA3 on VA.gov, or effectively people who have not verified their identity and people who have. I do think there might be some meaningful distinctions for LOA2 on other VA sites, but I don't think that affects how VA.gov works.

> 2.	When creating an account, do I use ID.Me, or something else?

Use ID.me.

**Comment by tressaellen** [comment]
Hello @joshkimux !
> Is there a design reason for not making the first of these two modal? 

 That was _should_ be a modal, I'm not sure why it's showing as a non-modal. @ajakabcin could you make a ticket for FE to fix that?

**Comment by AngelaFowler82** [comment]
Another question:

 



 

If we know what the issue is and how the user can fix it, why are we presenting the nonspecific “something went wrong” message? Those nonspecific messages are highly frustrating for the end user, and the more we can avoid them the better.

**Comment by Samara-Strauss** [comment]
> If we know what the issue is and how the user can fix it, why are we presenting the nonspecific “something went wrong” message? Those nonspecific messages are highly frustrating for the end user, and the more we can avoid them the better.

These are generally used when there's something the user can't fix, like a backend problem. Are you seeing a use case where you are getting this error for something you can fix? If so, we can fix that.

**Comment by AngelaFowler82** [comment]
Yes, my apologies. It occurs when a LOI1 user chooses “Go to my Profile” under “My VA.”

**Comment by AngelaFowler82** [comment]
Hello again,

This is a heads up that I was able to successfully update the direct deposit info for user27, the user whose account was flagged for fraud.

**Comment by Samara-Strauss** [comment]
> Yes, my apologies. It occurs when a LOA1 user chooses “Go to my Profile” under “My VA.”

@AngelaFowler82 Ah, thank you for clarifying. This is actually a known issue we need to address related to the fact that the code is trying to get some information for LOA1 users that we know we can't get. The ticket we have open for it is #19349. Once we fix the code so it's no longer trying to catch that information, this alert will go away.

> This is a heads up that I was able to successfully update the direct deposit info for user27, the user whose account was flagged for fraud.

I just checked, and I don't think this error is flagged for fraud anymore. I don't know that there is a way for us to turn that on easily. You can ignore that for now.

**Comment by AngelaFowler82** [comment]
Thank you for the info. Can I safely ignore all such findings at this point?

**Comment by Samara-Strauss** [comment]
> Can I safely ignore all such findings at this point?

Sorry, regarding which point?

**Comment by AngelaFowler82** [comment]
Situations in which the individual use cases don’t act the way the documentation says they should I.E. user 27.

**Comment by Samara-Strauss** [comment]
> Situations in which the individual use cases don’t act the way the documentation says they should I.E. user 27.

Please make a note in case it's indicative of a problem. But for user 27, yes, you can ignore that particular use case.

**Comment by joshkimux** [comment]
@AngelaFowler82 @Samara-Strauss , I've completed my review for Profile 2.0 which covers

- VO + Safari
- Axe checks
- Magnification
- Mobile
- Keyboard

Angela will handle the rest when it comes to testing with other screen readers 👍

**Comment by Samara-Strauss** [comment]
Great! Keep us posted on how it goes.

If there is work that comes out of this, can we expect that you all will create tickets outlining the problem and what needs to be fixed? Just want to make sure our expectations are correct. CC/ @ajakabcin

**Comment by AngelaFowler82** [comment]
Absolutely. It is our practice to create tickets describing all issues we find unless otherwise instructed.

**Comment by Samara-Strauss** [comment]
Great, thank you for letting me know!

**Comment by joshkimux** [comment]
@Samara-Strauss you'll find that most of the tickets we've attached to this epic provide directions on how to fix the problems they describe! If your team needs extra clarification or support, just let us know.

**Comment by Samara-Strauss** [comment]
> @Samara-Strauss you'll find that most of the tickets we've attached to this epic provide directions on how to fix the problems they describe! If your team needs extra clarification or support, just let us know.

I am so sorry —  I was viewing this in Github and didn't see the attached tickets. I see them now via Zenhub. Thank you for getting those together!

**Comment by joshkimux** [comment]
@Samara-Strauss Angela and I have completed this audit, it's ready for y'all to assign out! 🥳

**Comment by Samara-Strauss** [comment]
Great, thanks @joshkimux and @AngelaFowler82! @ajakabcin and I will review the tickets and triage accordingly. We really appreciate your help.

**Comment by ajakabcin** [comment]
I believe this should be in @SKasimow's backlog - tagging over now
